options/pin.go

```diff
@@ -1,13 +1,23 @@
 package options
 
+import "fmt"
+
 type PinAddSettings struct {
 	Recursive bool
 }
 
+type TypeSettings struct {
+	Type string
+}
+
 type PinLsSettings struct {
 	Type string
 }
 
+type PinIsPinnedSettings struct {
+	WithType string
+}
+
```

Comment:
It looks like PinLsSettings and PinIsPinnedSettings could be aliased to the same thing (was `TypeSettings` for this?) and have the same handling (currently duplicated code). Having `PinIsPinnedSettings.Type` rather than `WithType` for the name of the fields sounds fine, as meaning is pretty obvious anyways.